### PR TITLE
Allow setting active by deploy id as well as port

### PR DESCRIPTION
**Pull request by jsuresh**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -16,7 +16,13 @@ type Client interface {
 	Push() (string, error)
 	// Run runs the specified deploy, returning the port it is listening on.
 	Run(deployId string) (int, error)
+
+	// Sets active deploy by port
```
Comment (soulplant) [inline client.go]: Can be a TODO but it seems that SetActiveByPort() and SetActiveById() are better names than the current ones.

Comment (jsuresh) [inline client.go]: Good pick, will fixup in a follow up commit (This change will touch basically every file)

File: server.go [hunk 2]
```diff
@@ -409,6 +409,16 @@ func (s *ServerImpl) SetMainByPort(port int) error {
 	return s.reloadHaproxy(port)
 }
 
+func (s *ServerImpl) SetMainById(id string) error {
+	for port, deployId := range s.config.Ports {
+		if deployId == id {
+			return s.reloadHaproxy(port)
```
Comment (latos) [inline server.go]: Perhaps this should call SetMainByPort(port) instead

Then if we choose to implement a check that something's actually running on that port...

Comment (latos) [inline server.go]: The RPC request could have a boolean "force" value (set by the client based on flag) - if it's not true, then the request fails unless there's a 200 OK status server running on that port.

Comment (jsuresh) [inline server.go]: Good point, should have a single clear choke point. Added a issue to track this.

**Pull request merged**